### PR TITLE
Remove Spring Data maven-invoker style tests

### PR DESCRIPTION
**Pull request by geoand**
Unblocks: #7922

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

**Comment by gastaldi** [comment]
@geoand what's missing to make this a real PR?

File: test-framework/junit5-internal/src/main/java/io/quarkus/test/InMemoryLogHandler.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+
+public class InMemoryLogHandler extends Handler {
+
+    private final Predicate<LogRecord> predicate;
+
+    public InMemoryLogHandler(Predicate<LogRecord> predicate) {
+        this.predicate = predicate;
```
Comment (gastaldi) [inline test-framework/junit5-internal/src/main/java/io/quarkus/test/InMemoryLogHandler.java]:
```suggestion
import java.util.logging.LogRecord;
import org.wildfly.common.Assert;

public class InMemoryLogHandler extends Handler {

    private final Predicate<LogRecord> predicate;

    public InMemoryLogHandler(Predicate<LogRecord> predicate) {
        this.predicate = Assert.checkNotNullParam("predicate",predicate);
```

**Comment by geoand** [comment]
I just made it draft so that tests could run on my GH actions, no other reason :)

**Comment by gastaldi** [comment]
Cool, it looks like it's ready to be merged after squashed :wink:

**Comment by geoand** [comment]
I squashed you suggestion locally, but I am currently looking into a weird failure I got in CI (only on Windows of course...)

**Comment by gastaldi** [comment]
Yeah, I was about to ask about that, that's a weird error

**Comment by geoand** [comment]
Very weird indeed... But in order to avoid such flakes, I'll probably end up splitting this test from the rest

**Comment by geoand** [comment]
@gastaldi PR updated. Want to take a look and make sure you are OK with the new approach?

**Review by aureamunoz — APPROVED** [review]

**Comment by geoand** [comment]
@gastaldi CI (on my fork) passed with my latest change, so unless you have any objections, I'll merge when CI is done

**Comment by gastaldi** [comment]
No objections. Do it

**Pull request merged**